packages/cosmos_ui_components/lib/utils/group_by_extension.dart

```diff
@@ -0,0 +1,6 @@
+extension Iterables<E> on Iterable<E> {
+  Map<K, List<E>> groupBy<K>(K Function(E) keyFunction) => fold(
+        <K, List<E>>{},
+        (Map<K, List<E>> map, E element) => map..putIfAbsent(keyFunction(element), () => <E>[]).add(element),
+      );
```

Comment:
cool addition, could we actually put it in `cosmos_utils` so its available for broader use?